e2e/go.mod

```diff
@@ -1,79 +1,83 @@
 module github.com/openshift/cluster-capi-operator/e2e
 
-go 1.18
+go 1.21
+
+toolchain go1.21.5
 
 require (
-	github.com/aws/aws-sdk-go v1.44.213
+	github.com/aws/aws-sdk-go v1.44.298
 	github.com/google/go-cmp v0.6.0
-	github.com/onsi/ginkgo/v2 v2.13.0
-	github.com/onsi/gomega v1.28.1
+	github.com/onsi/ginkgo/v2 v2.15.0
+	github.com/onsi/gomega v1.31.1
 	github.com/openshift/api v0.0.0-20220921125526-1866ef90edbf
-	k8s.io/api v0.27.6
-	k8s.io/apimachinery v0.27.6
-	k8s.io/client-go v0.27.6
-	k8s.io/utils v0.0.0-20230209194617-a36077c30491
-	sigs.k8s.io/cluster-api v1.5.3
-	sigs.k8s.io/cluster-api-provider-aws/v2 v2.2.4
-	sigs.k8s.io/cluster-api-provider-gcp v1.5.0
-	sigs.k8s.io/cluster-api-provider-ibmcloud v0.6.0
-	sigs.k8s.io/controller-runtime v0.15.2
-	sigs.k8s.io/yaml v1.3.0
+	k8s.io/api v0.28.6
+	k8s.io/apimachinery v0.28.6
+	k8s.io/client-go v0.28.6
+	k8s.io/utils v0.0.0-20240102154912-e7106e64919e
+	sigs.k8s.io/cluster-api v1.6.2
+	sigs.k8s.io/cluster-api-provider-aws/v2 v2.3.5
```

Comment:
This is still based on the CAPI 1.5.x CAPA release, will update it once 2.4 is out and we did the rebase for the provider fork.